pkg/vcs/vcs.go

```diff
@@ -58,3 +87,86 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App) (string, string, error) {
+	vcs := app.Status.Staged.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" {
+		return "", "", fmt.Errorf("app has no acornfile information in vcs")
+	}
+
+	// Create auth object to use when fetching and cloning git repos
+	auth, err := ssh.NewSSHAgentAuth("git")
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Since we use ssh auth to clone the repo we need a git url but will sometimes get http urls
+		var gitUrl string
+		httpUrl, err := url.Parse(remote)
+		if err == nil {
+			gitUrl = fmt.Sprintf("git@%s:%s", httpUrl.Host, httpUrl.Path[1:])
+		} else {
+			gitUrl = remote
+		}
+
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote) {
```

Comment:
I don't think it is possible for `idx >= len(remote)`. Perhaps `idx >= len(remote) - 1`?